File: api/Ganymede.Api.Data/Monsters/BasicStats.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Ganymede.Api.Data.Monsters
{
    public class BasicStats
    {
        public int ID { get; set; }
        public float CR { get; set; }
        public int XP { get; set; }
        public int Strength { get; set; }
        public int Dexterity { get; set; }
        public int Constitution { get; set; }
        public int Intelligence { get; set; }
        public int Wisdom { get; set; }
        public int Charisma { get; set; }

        public int MonsterID { get; set; }
        [ForeignKey("MonsterID")]
        public Monster Monster { get; set; }
    }
}

File: src/app/campaign/modules/monster/models/basicStats.ts
export class BasicStats {
	public id: number;
	public cr: number;
	public xp: number;
	public strength: number;
	public dexterity: number;
	public constitution: number;
	public intelligence: number;
	public wisdom: number;
	public charisma: number;

	public static getDefault(): BasicStats {
		const stats = new BasicStats();
		stats.id = -1;
		return stats;
	}

	public static isEqual(a: BasicStats, b: BasicStats): boolean {
		if ((!a && b) || (a && !b)) return false;
		if (!a && !b) return true;

		return (
			a === b ||
			(a.id === b.id &&
				a.cr === b.cr &&
				a.xp === b.xp &&
				a.strength === b.strength &&
				a.dexterity === b.dexterity &&
				a.constitution === b.constitution &&
				a.intelligence === b.intelligence &&
				a.wisdom === b.wisdom &&
				a.charisma === b.charisma)
		);
	}
}
